### ISSUE TITLE
[ios] ContentPage.ToolbarItems is not currently being Garbage Collected (clicked event handler used)

### ISSUE THREAD

**Issue by nevse** [dotnet/maui#22814]
### Description

Hi

With the following Xaml on a ContentPage, the page doesn't get garbage collected when navigating back:
```xaml
    <ContentPage.ToolbarItems>
        <ToolbarItem Text="Test" Clicked="OnSomeItemClicked" />
    </ContentPage.ToolbarItems>
```
I used packages from that fix https://github.com/dotnet/maui/issues/20703 (8.0.60-ci.net8.24302.1) and it stil doesn't work properly. 

### Steps to Reproduce

Use example from https://github.com/nevse/maui-tests-toolbaritemleak to reproduce the problem.
Click on "show page with leak" button, navigate back and press "check leak" button. 

### Link to public reproduction project repository

https://github.com/nevse/maui-tests-toolbaritemleak

### Version with bug

Nightly / CI build (Please specify exact version)

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

**Comment by jsuarezruiz** [comment]
cc @jonathanpeppers

**Comment by jonathanpeppers** [comment]
@nevse if you make `OnSomeItemClicked` `static`, does the problem go away?

**Comment by brentpbc** [comment]
I just tested and can confirm that changing the clicked method to be a static method does indeed prevent the leak on iOS device. 
<img width="269" alt="Screenshot 2024-06-05 at 12 08 29 pm" src="https://github.com/dotnet/maui/assets/55366323/d77e98f6-fddd-44de-af50-207d8bf25857">

See attached test app(See Views/ToolBarContentPage and Views/ToolbarStaticContentPage).

[MemoryTestApp 20240605_toolbar.zip](https://github.com/user-attachments/files/15572267/MemoryTestApp.20240605_toolbar.zip)

**Comment by williambohrmann3** [comment]
I am able to reproduce (app crashes) in my [repository](https://github.com/williambohrmann3/GeoViewAppCrashMaui) when subscribing to `Clicked` event for toolbar item. Per suggestion, marking event handler as static prevents app crash. All testing completed with latest MAUI nightly build.

**Comment by williambohrmann3** [comment]
Thanks for fixing, tested in my isolated reproducer and our production app :)

**Comment by brentpbc** [comment]
Thanks everyone, I can also confirm it's fixed in 8.0.60 in my test reproduction app when using a non static clicked event on the ToolbarItem.
